packages/mobile/src/components/composer-input/ComposerInput.tsx

```diff
@@ -298,16 +299,21 @@ export const ComposerInput = forwardRef(function ComposerInput(
   )
 
   const handleSubmit = useCallback(() => {
-    const mentions =
-      getUserMentions(value)?.map((match) => {
-        return {
-          handle: match.text.replace('@', ''),
-          userId: userIdMap[match.text]
-        }
-      }) ?? []
-
-    onSubmit?.(restoreLinks(value), mentions)
-  }, [getUserMentions, onSubmit, restoreLinks, userIdMap, value])
+    if (internalRef.current) {
+      internalRef.current.blur()
+      setTimeout(() => {
+        setValue('')
+        const mentions =
+          getUserMentions(latestValueRef.current)?.map((match) => {
+            return {
+              handle: match.text.replace('@', ''),
+              userId: userIdMap[match.text]
+            }
+          }) ?? []
+        onSubmit?.(restoreLinks(latestValueRef.current), mentions)
```

Comment:
probably dont need to do this stuff inside of the timeout func? Do you need this here for a reason?

Reply:
i think it's necessary to wait for the latestValueRef to be autocorrected after the blur

Reply:
gotcha 👍